gensim/models/atmodel.py

```diff
@@ -882,6 +882,84 @@ def get_document_topics(self, word_id, minimum_probability=None):
 
         raise NotImplementedError('Method "get_document_topics" is not valid for the author-topic model. Use the "get_author_topics" method.')
 
+    def get_new_author_topics(self, corpus, minimum_probability=None):
+        """Infers topics for new author.
+
+        Infers a topic distribution for a new author over the passed corpus of docs,
+        assuming that all documents are from this single new author.
+
+        Args:
+            corpus (list): Bag-of-words representation of the documents to get topics for.
+            minimum_probability (float): Ignore topics with probability below this value
+                (None by default). If set to None, a value of 1e-8 is used to prevent 0s.
+        Returns:
+            topic distribution for the given list of documents `corpus` in bow format,
+            as a list of `(topic_id, topic_probability)` 2-tuples.
+        """
+
+        # TODO: how should this function look like for get_new_author_topics?
+        def rho():
+            return pow(self.offset + 1 + 1, -self.decay)
+
+        # Wrap in fuction to avoid code duplication.
+        def rollback_new_author_chages():
+            self.state.gamma = self.state.gamma[0:-1]
+
+            del self.author2doc[new_author_name]
+            a_id = self.author2id[new_author_name]
+            del self.id2author[a_id]
+            del self.author2id[new_author_name]
+
+            for new_doc_id in corpus_doc_idx:
+                del self.doc2author[new_doc_id]
+
+            self.num_authors -= num_new_authors
+
+        try:
+            len_input_corpus = len(corpus)
+        except TypeError:
+            logger.warning("input corpus stream has no len(); counting documents")
+            len_input_corpus = sum(1 for _ in corpus)
+        if len_input_corpus == 0:
+            logger.warning("AuthorTopicModel.get_new_author_topics() called with an empty corpus")
+            return
```

Comment:
Why "empty return" needed? Maybe better raise exception explicitly (for all empty `return`)